flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/MockIndexedInputGate.java

```diff
@@ -63,6 +64,9 @@ public void requestPartitions() {}
     @Override
     public void resumeConsumption(InputChannelInfo channelInfo) {}
 
+    @Override
+    public void acknowledgeAllRecordsProcessed(InputChannelInfo channelInfo) throws IOException {}
```

Comment:
How about throwing an `UnsupportedOperationException`? I feel it would be safer.